In [3]:
import pandas as pd
import numpy as np

df_orders = pd.read_csv(r'C:\Users\Mateus Fernandes\Documents\olist-analytics\data\raw\olist_orders_dataset.csv')

print("Dados carregados com sucesso!!! Verificação!")
print(f"Linhas: {df_orders.shape[0]}")
print(f"Colunas: {df_orders.shape[1]}")
print(f"Valores tupla: {df_orders.shape}")


Dados carregados com sucesso!!! Verificação!
Linhas: 99441
Colunas: 8
Valores tupla: (99441, 8)
Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date'],
      dtype='str')


In [5]:
df_orders['order_purchase_timestamp'] = pd.to_datetime(df_orders['order_purchase_timestamp'])
df_orders['order_delivered_customer_date'] = pd.to_datetime(df_orders['order_delivered_customer_date'])
df_orders['order_estimated_delivery_date'] = pd.to_datetime(df_orders['order_estimated_delivery_date'])

print("Colunas convertidas!!")

# Verificar os tipos
print(df_orders[['order_purchase_timestamp', 'order_delivered_customer_date']].dtypes)



Colunas convertidas!!
order_purchase_timestamp         datetime64[us]
order_delivered_customer_date    datetime64[us]
dtype: object


In [6]:
# Calcular a diferença em dias
df_orders['delivery_time'] = (df_orders['order_delivered_customer_date'] - df_orders['order_purchase_timestamp']).dt.days

# Ver as primeiras linhas com a nova coluna
df_orders[['order_purchase_timestamp', 'order_delivered_customer_date', 'delivery_time']].head()

,order_purchase_timestamp,order_delivered_customer_date,delivery_time
0,2017-10-02 10:56:33,2017-10-10 21:25:13,8.0
1,2018-07-24 20:41:37,2018-08-07 15:27:45,13.0
2,2018-08-08 08:38:49,2018-08-17 18:06:29,9.0
3,2017-11-18 19:28:06,2017-12-02 00:28:42,13.0
4,2018-02-13 21:18:39,2018-02-16 18:17:02,2.0


In [7]:
print("=== ESTATÍSTICAS DO TEMPO DE ENTREGA ===")
print(df_orders['delivery_time'].describe())

print(f"\nValores nulos em delivery_time: {df_orders['delivery_time'].isnull().sum()}")

=== ESTATÍSTICAS DO TEMPO DE ENTREGA ===
count    96476.000000
mean        12.094086
std          9.551746
min          0.000000
25%          6.000000
50%         10.000000
75%         15.000000
max        209.000000
Name: delivery_time, dtype: float64

Valores nulos em delivery_time: 2965


## 📊 Limpeza de Dados - olist_orders_dataset.csv

### Conversões Realizadas
- `order_purchase_timestamp` → datetime
- `order_delivered_customer_date` → datetime
- `order_estimated_delivery_date` → datetime

### Nova Coluna Criada
- **delivery_time**: tempo em dias entre a compra e a entrega

### Estatísticas do Tempo de Entrega
- **Total de pedidos analisados**: 96.476
- **Média**: 12 dias
- **Mediana**: 10 dias
- **Mínimo**: 0 dias (entrega no mesmo dia)
- **Máximo**: 209 dias
- **Desvio padrão**: 9.5 dias

### Dados Faltantes
- **2.965 pedidos** sem data de entrega (ainda não entregues)

### Observações
- A maioria dos pedidos é entregue entre 6 e 15 dias
- O tempo médio de entrega é de 12 dias
- Os dados faltantes representam pedidos que ainda não foram entregues

### Próximos Passos
- Análise de correlação entre tempo de entrega e nota do cliente
- Análise de sazonalidade das vendas
